components/Animation.js

```diff
@@ -0,0 +1,45 @@
+const React = require('react');
+
+class Animation extends React.Component {
+
+  // TODO: create a componentWillUpdate() here
+  componentWillUpdate() {
+    this.showLoadingBar();
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      url: ' http://placehold.it/500x150'
+    };
+    this.showLoadingBar = this.showLoadingBar.bind(this);
+  }
+
+  getNewCat() {
+    fetch('http://api.giphy.com/v1/gifs/random?api_key=dc6zaTOxFJmzC')
+      .then((res, err) => { 
+        if (err) {
+          console.log('Something went wrong with fetching your new cat!', err)
+        } else {
+          res.json().then(result => this.setState({ url: result.data.fixed_height_downsampled_url }));
+        }
+      });
+  }
+
+  showLoadingBar() {
+    const progressBar = document.getElementById('progress-bar');
+    progressBar.className = 'off on';
+    setTimeout(() => progressBar.className = 'off', 3000);
+  }
+
+  render() {
+    return (
+      <div>
+        <img src={this.state.url} height="100px"/>
+        <div><button onClick={() => this.getNewCat()}>New random .gif!</button></div>
```

Comment:
The `onClick` prop doesn't need to be an arrow function, we can just pass the reference to `this.getNewCat` instead. Make sure it's bound in the constructor, though!